apps/web/pages/api/addresses/[address]/usernamesDiscountMessage/index.ts

```diff
@@ -0,0 +1,112 @@
+import type { NextApiRequest, NextApiResponse } from 'next';
+import { kv } from '@vercel/kv';
+import { hasRegisteredWithDiscount } from 'apps/web/src/components/Usernames/registrarController';
+import { ethers } from 'ethers';
+import {
+  trustedSignerAddress,
+  trustedSignerPKey,
+} from 'apps/web/src/components/Usernames/constants';
+
+type PreviousClaim = {
+  address: string;
+  signedMessage: string;
+};
+
+type LinkedAddresses = {
+  idemKey: string;
+  linkedAddresses: string[];
+};
+
+const expiry = 300; //  5 minutes in seconds
```

Comment:
nit: could consider making this an env var for easier updates as I could see us wanting to tweak this once we start dogfooding.